web-registry/src/lib/ecocredit/api.ts

```diff
@@ -217,6 +206,70 @@ export const getEcocreditTxs = async (): Promise<TxResponse[]> => {
   });
 };
 
+export const getBridgedEcocreditsForAccount = async (
+  addr?: string,
+  sanityCreditClassData?: AllCreditClassQuery,
+): Promise<BridgedEcocredits[] | undefined> => {
+  if (addr) {
+    // Since bridged ecocredits are canceled on Regen Ledger,
+    // the only way to get them is to fetch txs by event and loop through the tx messages.
+    // Txs can have multiple messages and MsgBridge messages can have multiple credits
+    // so we can't really support pagination.
+    const res = await getTxsByEvent({
+      events: [
+        `${messageActionEquals}'/${MsgBridge.$type}'`,
+        `message.sender='${addr}'`,
+      ],
+      orderBy: OrderBy.ORDER_BY_DESC,
+    });
+
+    const bridgedCredits: BridgedEcocredits[] = [];
+    const batchesMap = new Map<string, IBatchInfoWithClassProject>();
+    for (let i = 0; i < res.txs.length; i++) {
```

Comment:
I'm wondering if instead of having 3 nested for-loop we could have done 2 map and a forEach to increase readability.
```
const messages = res.txs.map(tx => ...)
const credits = messages.map(message => ...)
const credits.forEach(credit => ...) 
```

Reply:
All good then was just a thought, thanks for sharing that.